core-web/libs/block-editor/src/lib/shared/services/ai-content/ai-content.service.ts

```diff
@@ -0,0 +1,75 @@
+import { Observable, throwError } from 'rxjs';
+
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { catchError, map } from 'rxjs/operators';
+
+const API_KEY = '';
+
+interface Message {
+    role: string;
+    content: string;
+}
+
+interface Choice {
+    message: Message;
+    finish_reason: string;
+}
+
+interface OpenAIResponse {
+    id: string;
+    object: string;
+    created: number;
+    model: string;
+    usage: {
+        prompt_tokens: number;
+        completion_tokens: number;
+        total_tokens: number;
+    };
+    choices: Choice[];
+}
+
+@Injectable({
+    providedIn: 'root'
+})
+export class AiContentService {
+    constructor(private http: HttpClient) {}
+
+    getIAContent(prompt: string): Observable<string> {
+        const body = JSON.stringify({
+            model: 'gpt-3.5-turbo',
```

Comment:
Maybe this can be a const

Reply:
this will be removed soon, we will integrate with a backend. instead of directly to the openai.